### PR TITLE
Added declarations file addressing issue#12 for package rclone.js

### PR DESCRIPTION
**Pull request by atifcppprogrammer**
### Template
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Checks for adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### Acknowledgements
Special thanks to **[jsejcksn](https://github.com/jsejcksn)** for providing crucial help in producing types for this package see [this](https://stackoverflow.com/questions/72939152/adding-dynamic-properties-to-typescript-namespace-in-a-declarations-file) for context.

### CODE REVIEW

File: types/rclone.js/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictFunctionTypes": true,
+        "strictNullChecks": true,
+        "esModuleInterop": true,
```
Comment (peterblazejewicz) [inline types/rclone.js/tsconfig.json:11]:
```suggestion
```
https://github.com/DefinitelyTyped/DefinitelyTyped#esmoduleinteropallowsyntheticdefaultimports
you would have to amend your imports in tests

File: types/rclone.js/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
+// Type definitions for rclone.js 0.6
+// Project: rclone.js
+// Definitions by: atifcppprogrammer <https://github.com/atifcppprogrammer>, sntran <https://github.com/sntran>
```
Comment (peterblazejewicz) [inline types/rclone.js/index.d.ts:3]:
```suggestion
// Definitions by: atifcppprogrammer <https://github.com/atifcppprogrammer>
//                 sntran <https://github.com/sntran>
```

File: types/rclone.js/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+import type { ChildProcess } from 'node:child_process';
```
Comment (peterblazejewicz) [inline types/rclone.js/index.d.ts:7]:
```suggestion

import type { ChildProcess } from 'node:child_process';
```
just for readability

File: types/rclone.js/rclone.js-tests.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import rclone, { promises as rclonePromise } from "rclone.js";
```
Comment (peterblazejewicz) [inline types/rclone.js/rclone.js-tests.ts:1]:
```suggestion
import rclone = require("rclone.js");
import { promises as rclonePromise } from "rclone.js";
```
or use `import * as rclone` syntax

File: types/rclone.js/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
+// Type definitions for rclone.js 0.6
+// Project: rclone.js
```
Comment (peterblazejewicz) [inline types/rclone.js/index.d.ts:2]:
```suggestion
// Project: https://github.com/sntran/rclone.js
```

Comment (atifcppprogrammer) [reply types/rclone.js/index.d.ts:2]: Addressed in latest commit

File: types/rclone.js/rclone.js-tests.ts (hunk 4 above)
Comment (atifcppprogrammer) [reply types/rclone.js/rclone.js-tests.ts:1]: Addressed in latest commit

File: types/rclone.js/index.d.ts (hunk 3 above)
Comment (atifcppprogrammer) [reply types/rclone.js/index.d.ts:9]: Addressed in latest commit

File: types/rclone.js/tsconfig.json (hunk 1 above)
Comment (atifcppprogrammer) [reply types/rclone.js/tsconfig.json]: Addressed in latest commit

File: types/rclone.js/index.d.ts (hunk 2 above)
Comment (atifcppprogrammer) [reply types/rclone.js/index.d.ts]: Addressed in latest commit

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@atifcppprogrammer thanks!

**Comment by atifcppprogrammer** [comment]
Ready to merge

**Pull request merged**